### PR TITLE
#1726 - Deploy to Prod from CD Pipeline

### PR DESCRIPTION
**Pull request by ldraney**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

issue #1726 

This is a simple but important change to our pipeline to finally deploy to production. 
Upon merge of this PR, we will begin using this pipeline for our deployment process. 

This PR has the enhancement label, so it will semver bump a minor version.  This is because the plan is to move from v1.7.12 to 1.8.0.  I tested this transition, as well as the new code, in my POC repo; please see the testing section below. 

In preparation for merging this PR, all non-"v" tags were deleted, tag 1.7.12 was created, and the RELEASE_VERSION repo variable was adjusted to 1.7.12.  

This PR also fixes a bug where deploys sent a SHA instead of the tag.  [Here's an explanation. ](https://docs.google.com/document/d/1uHoozVFtNZNIAr0ixcVvooOMVgqOHTuxwmZVM1sjkRY/edit?usp=sharing)

## Type of change

Please check the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

How has this been tested? (e.g. Unit tests, Tested locally, Tested as a GitHub action, Depoyed to dev, etc)  
Please provide relevant information and / or links to demonstrate the functionality or fix. (e.g. screenshots, link to deployment, regression test results, etc)

[POC to deploy to prod](https://github.com/dialectic-devops/demo-api/actions/runs/9352832789) 
To verify the code change, please go to the [.github/workflows directory](https://github.com/dialectic-devops/demo-api/tree/main/.github/workflows) and view the cd-pipeline.yml. 

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] The ticket is now moved into the DEV test column

### CODE REVIEW

File: .github/workflows/cd-pipeline.yml [hunk 1]
```diff
@@ -129,3 +129,12 @@ jobs:
     with:
       draftReleaseReference: ${{ needs.create-release-notes.outputs.draftReleaseReference }}
 
+  deploy-to-prod:
+    needs: [publish-release-notes, create-and-post-tag]
```
Comment (kalbfled) [inline .github/workflows/cd-pipeline.yml:133]: ditto

File: Makefile [hunk 2]
```diff
@@ -32,8 +32,10 @@ install-safety:
 
 check-dependencies: install-safety ## Scan dependencies for security vulnerabilities
 	# 12 Dec 2023: 51668 is fixed with >= 2.0.0b1 of SQLAlchemy. Ongoing refactor to upgrade.
+	# 6 June 2024: 70624 vulnerability found with flask-cors version 4.0.0 
+	# 6 June 2024: 70612 vulnerability found with jinja2 version 3.1.3
```
Comment (kalbfled) [inline Makefile:36]: If a version exists that clears a vulnerability, please annotate the version here.

File: .github/workflows/cd-pipeline.yml [hunk 3]
```diff
@@ -105,7 +105,7 @@ jobs:
       previousVersion: ${{ needs.create-and-post-tag.outputs.previousVersion }}
 
   deploy-to-staging:
-    needs: create-release-notes
+    needs: [create-release-notes, create-and-post-tag]
```
[the remark this answers is not in the document]
Comment (cris-oddball) [reply .github/workflows/cd-pipeline.yml:108]: It is necessary to acquire the tags, apparently, he walked me through this in his POC.

File: .github/workflows/cd-pipeline.yml (hunk 1 above)
Comment (cris-oddball) [reply .github/workflows/cd-pipeline.yml:133]: ditto re he walked me through this in his POC

File: Makefile (hunk 2 above)
Comment (kalbfled) [reply Makefile:36]: We discussed this.

**Review by kalbfled — APPROVED** [review]

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**